api/gateway-operator/v1beta1/dataplane_types.go

```diff
@@ -240,6 +241,22 @@ type DataPlaneServicePort struct {
 	// More info: https://kubernetes.io/docs/concepts/services-networking/service/#defining-a-service
 	// +optional
 	TargetPort intstr.IntOrString `json:"targetPort,omitempty"`
+
+	// The port on each node on which this service is exposed when type is
+	// NodePort or LoadBalancer. Usually assigned by the system. If a value is
```

Comment:
Maybe it's worth noting that handling this `NodePort` setting for the `LoadBalancer` type depends on the provider (operator) that handles load balancers. In some cases, they don't use node ports at all. WDYT?

Reply:
This was copy pasted from upstream so I believe we might leave as is. WDYT?